### PR TITLE
feat(#6): Checks index and unstaged files

### PR DESCRIPTION
**Pull request by lordofthejars**

### CODE REVIEW

**Comment by MatousJobanek** [comment]
LGTM

**Comment by bartoszmajsak** [comment]
I will review as soon as I arrive. Great stuff!

File: strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java [hunk 1]
```diff
@@ -12,19 +18,18 @@
 import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
 import org.eclipse.jgit.treewalk.CanonicalTreeParser;
 
-class GitDiffFetcher {
+class GitFetcher {
```
Comment (bartoszmajsak) [inline strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java]: I think we can have a better name here. I didn't like `GitDiffFetcher`, but `GitFetcher` is potentially more confusing, as there is `git fetch` operation :) This component now does not only deals with diffs, but all changes, so I would propose `GitChangeResolver`?

File: strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java [hunk 2]
```diff
@@ -41,4 +46,33 @@
...
+             Git git = new Git(repository)) {
+
+            return getNotCommittedFiles(git.status().call());
+
+        } catch (IOException | GitAPIException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private Set<String> getNotCommittedFiles(final Status status) {
```
Comment (bartoszmajsak) [inline strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java]: Maybe `Uncommitted`?

File: strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java [hunk 3]
```diff
@@ -41,4 +46,33 @@
...
+        notCommittedFiles.addAll(status.getUntracked());
+        notCommittedFiles.addAll(status.getModified());
+
+        return notCommittedFiles;
+
+    }
+
+    private Repository getRepository() throws IOException {
+        final FileRepositoryBuilder builder = new FileRepositoryBuilder();
+
```
Comment (bartoszmajsak) [inline strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitFetcher.java]: I am not sure if we need all those empty lines - can we clean it up?

File: strategies/changed/src/test/java/org/arquillian/smart/testing/vcs/git/GitFetcherTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.arquillian.smart.testing.vcs.git.GitRepositoryUnpacker.unpackRepository;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GitFetcherTest {
```
Comment (bartoszmajsak) [inline strategies/changed/src/test/java/org/arquillian/smart/testing/vcs/git/GitFetcherTest.java]: Is it a total rewrite of `GitDiffFetcherTest` or we just have odd diff here? :)

Comment (lordofthejars) [inline strategies/changed/src/test/java/org/arquillian/smart/testing/vcs/git/GitFetcherTest.java]: No just the new bits

Comment (bartoszmajsak) [inline strategies/changed/src/test/java/org/arquillian/smart/testing/vcs/git/GitFetcherTest.java]: Then something is odd with commits / github diff - but that's fine. Just curious

File: strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitChangeResolver.java [hunk 5]
```diff
@@ -47,19 +47,19 @@
...
-            return getNotCommittedFiles(git.status().call());
+            return getUnCommittedFiles(git.status().call());
 
         } catch (IOException | GitAPIException e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private Set<String> getNotCommittedFiles(final Status status) {
+    private Set<String> getUnCommittedFiles(final Status status) {
```
Comment (bartoszmajsak) [inline strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitChangeResolver.java]: I'm a pita, but it's `Uncommitted` (lowercase `c`)

Comment (lordofthejars) [inline strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitChangeResolver.java]: No worries actually I thought it was two words

**Pull request merged**